subprojects/packagefiles/mimalloc/meson_options.txt

```diff
@@ -0,0 +1,18 @@
+option(
+    'override',
+    type: 'boolean',
+    value: true,
```

Comment:
This default seems a bit surprising: merely depending on mimalloc (including indirectly through another dependency) will change global behavior.

Reply:
That's the opposite of surprising I'd say - the main point of this is that it is plug-and-play, using it explicitly is the exception.

Reply:
It may not be surprising for an allocator but it's surprising for a dependency.  In other circumstances, if I write `dependency('foo')`, the behavior of my program won't change unless I explicitly invoke foo.  More to the point, if my dependency `bar` writes `dependency('mimalloc')` without my knowledge, my project will be directly affected.  I'm not categorically opposed to this, but there's some value in requiring someone in the dependency chain to explicitly enable override.

Reply:
Ah - maybe you misunderstood this. It doesn't override malloc with just `dependecy('mimalloc')`, you need to actually link it. Sure, if a dependency declares its dependency with mimalloc as public dependency then you might be affected but this has to be quite intentional.